src/Concordium/Client/Runner/Helper.hs

```diff
@@ -132,3 +135,37 @@ value s =
 -- |Extract a value from the response. This assumes that the response from a GRPC call has a @value@ field.
 getValue :: forall a b. (Field.HasField a "value" b) => SimpleGetter (GRPCResponse a) (GRPCResponse b)
 getValue = to (fmap (^. CF.value))
+
+-- |Extract the response value of a GRPCResult and return it under the
+-- provided mapping.
+-- Returns @Left@ wrapping an error string describing its nature if the
+-- result contains an error, or a @Right@ wrapping the response value
+-- under the provided mapping otherwise.
+extractResponseValue :: (a -> b) -> GRPCResult (Either String a) -> Either String b
+extractResponseValue f res =
+  case res of
+    Left err -> Left $ "A GRPC error occurred: " <> err
+    Right resp ->
+      case grpcResponseVal resp of
+        Left err -> Left $ "Unable to convert response payload: " <> err
+        Right val -> Right $ f val
+
+-- |Extract the response value of a GRPCResult and return it under the
+-- provided mapping or fail with an error message if the result contains
+-- an error. Takes a string to be prepended to the error message.
+extractResponseValueOrFail :: (MonadIO m)
+  => (a -> b) -- |Result mapping
```

Comment:
`-- |` refers to the item below, so this will be shown for the `String` parameter. Move the comment or use ` -- ^ blah` instead 😊

Reply:
Thanks. So we use haddock documentation?

Do I read [this](http://haskell-haddock.readthedocs.io) to brush up on comment guidelines?

Reply:
Right, I guess it is something you pick up over time via reviews.
We don't have any clear guidelines, but extrapolating from our Haskell code is not always the best idea, as it is often lacking in documentation.

But reading a bit about the haddock syntax (even though we don't host an internal haddock yet) is useful.
Besides that, I try to convey things that would be useful for someone calling the function so that they don't need to look at the code.

Currently, I follow this rough template (private functions may naturally have less info)

```
One line that explains what the function does.

Perhaps more details about the functionality.

Preconditions (if any)

What it returns (if not obvious)
 - Including when it will logFatal / panic etc.
```

Reply:
Thanks for taking your time to elaborate on this. I think the template forms a great basis, and I saved it for future reference. I added a TODO on Zenhub which serves to ensure that all added docs follow best-practices. Once this PR (and its dependency in this repository) is merged into the feature branch I will make a pass to ensure this is the case.